File: src/client/acontext-py/src/acontext/resources/spaces.py
"""
Spaces endpoints.
"""

from collections.abc import Mapping, MutableMapping
from typing import Any

from ..client_types import RequesterProtocol


class SpacesAPI:
    def __init__(self, requester: RequesterProtocol) -> None:
        self._requester = requester

    def list(self) -> Any:
        return self._requester.request("GET", "/space")

    def status(self) -> Any:
        return self._requester.request("GET", "/space/status")

    def create(self, *, configs: Mapping[str, Any] | MutableMapping[str, Any] | None = None) -> Any:
        payload: dict[str, Any] = {}
        if configs is not None:
            payload["configs"] = configs
        return self._requester.request("POST", "/space", json_data=payload)

    def delete(self, space_id: str) -> None:
        self._requester.request("DELETE", f"/space/{space_id}")

    def update_configs(
        self,
        space_id: str,
        *,
        configs: Mapping[str, Any] | MutableMapping[str, Any],
    ) -> None:
        payload = {"configs": configs}
        self._requester.request("PUT", f"/space/{space_id}/configs", json_data=payload)

    def get_configs(self, space_id: str) -> Any:
        return self._requester.request("GET", f"/space/{space_id}/configs")

    def get_semantic_answer(self, space_id: str, *, query: str) -> Any:
        if not query:
            raise ValueError("query is required")
        params = {"query": query}
        return self._requester.request("GET", f"/space/{space_id}/semantic_answer", params=params)

    def get_semantic_global(self, space_id: str, *, query: str) -> Any:
        if not query:
            raise ValueError("query is required")
        params = {"query": query}
        return self._requester.request("GET", f"/space/{space_id}/semantic_global", params=params)

    def get_semantic_grep(self, space_id: str, *, query: str) -> Any:
        if not query:
            raise ValueError("query is required")
        params = {"query": query}
        return self._requester.request("GET", f"/space/{space_id}/semantic_grep", params=params)

File: src/client/acontext-ts/src/resources/spaces.ts
/**
 * Spaces endpoints.
 */

import { RequesterProtocol } from '../client-types';
import { buildParams } from '../utils';
import { ListSpacesOutput, ListSpacesOutputSchema, Space, SpaceSchema } from '../types';

export class SpacesAPI {
  constructor(private requester: RequesterProtocol) {}

  async list(options?: {
    limit?: number | null;
    cursor?: string | null;
    timeDesc?: boolean | null;
  }): Promise<ListSpacesOutput> {
    const params = buildParams({
      limit: options?.limit ?? null,
      cursor: options?.cursor ?? null,
      time_desc: options?.timeDesc ?? null,
    });
    const data = await this.requester.request('GET', '/space', {
      params: Object.keys(params).length > 0 ? params : undefined,
    });
    return ListSpacesOutputSchema.parse(data);
  }

  async create(options?: {
    configs?: Record<string, unknown>;
  }): Promise<Space> {
    const payload: Record<string, unknown> = {};
    if (options?.configs !== undefined) {
      payload.configs = options.configs;
    }
    const data = await this.requester.request('POST', '/space', {
      jsonData: Object.keys(payload).length > 0 ? payload : undefined,
    });
    return SpaceSchema.parse(data);
  }

  async delete(spaceId: string): Promise<void> {
    await this.requester.request('DELETE', `/space/${spaceId}`);
  }

  async updateConfigs(
    spaceId: string,
    options: {
      configs: Record<string, unknown>;
    }
  ): Promise<void> {
    const payload = { configs: options.configs };
    await this.requester.request('PUT', `/space/${spaceId}/configs`, {
      jsonData: payload,
    });
  }

  async getConfigs(spaceId: string): Promise<Space> {
    const data = await this.requester.request('GET', `/space/${spaceId}/configs`);
    return SpaceSchema.parse(data);
  }
}

